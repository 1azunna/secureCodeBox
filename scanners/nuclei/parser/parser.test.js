// SPDX-FileCopyrightText: 2021 iteratec GmbH
//
// SPDX-License-Identifier: Apache-2.0

const fs = require("fs");
const util = require("util");
// const {
//   validateParser,
// } = require("@securecodebox/parser-sdk-nodejs/parser-utils");

// eslint-disable-next-line security/detect-non-literal-fs-filename
const readFile = util.promisify(fs.readFile);

const { parse } = require("./parser");

test("parses empty result correctly", async () => {
  const fileContent = 
    await readFile(
      __dirname + "/__testFiles__/empty-test.jsonl",
      {
        encoding: "utf8",
      }
    )
  
  const findings = await parse(fileContent);
  //await expect(validateParser(findings)).resolves.toBeUndefined();
  expect(findings).toMatchSnapshot();
});

test("parses a single result correctly", async () => {
  const fileContent = 
    await readFile(
      __dirname + "/__testFiles__/single-test.jsonl",
      {
        encoding: "utf8",
      }
    )
  
  const findings = await parse(fileContent);
  //await expect(validateParser(findings)).resolves.toBeUndefined();
  expect(findings).toMatchSnapshot();
});

test("parses the dns result correctly", async () => {
  const fileContent = 
    await readFile(
      __dirname + "/__testFiles__/dns-test.jsonl",
      {
        encoding: "utf8",
      }
    )
  
  const findings = await parse(fileContent);
  //await expect(validateParser(findings)).resolves.toBeUndefined();
  expect(findings).toMatchSnapshot();
});

test("parses ftp result correctly", async () => {
  const fileContent = 
    await readFile(
      __dirname + "/__testFiles__/ftp-test.jsonl",
      {
        encoding: "utf8",
      }
    )
  
  const findings = await parse(fileContent);
  //await expect(validateParser(findings)).resolves.toBeUndefined();
  expect(findings).toMatchSnapshot();
});