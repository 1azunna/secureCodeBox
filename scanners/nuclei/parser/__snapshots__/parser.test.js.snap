// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parses empty result correctly 1`] = `Array []`;

exports[`parses ftp result correctly 1`] = `
Array [
  Object {
    "attributes": Object {
      "author": null,
      "extracted_results": null,
      "ip": "127.0.0.1",
      "matched": "www.example.com:21",
      "matcher_name": null,
      "metadata": Object {
        "password": "default",
        "username": "anonymous",
      },
      "reference": null,
      "tags": null,
      "timestamp": "2021-08-20T22:00:48.088618+02:00",
      "type": "network",
    },
    "category": "ftp-default-credentials",
    "description": "The name of the nuclei rule which triggered the finding: ftp-default-credentials",
    "location": "www.example.com",
    "name": "FTP Service with default credentials",
    "severity": "HIGH",
  },
  Object {
    "attributes": Object {
      "author": null,
      "extracted_results": null,
      "ip": "127.0.0.1",
      "matched": "www.example.com:21",
      "matcher_name": null,
      "metadata": Object {
        "password": "stingray",
        "username": "anonymous",
      },
      "reference": null,
      "tags": null,
      "timestamp": "2021-08-20T22:00:48.161349+02:00",
      "type": "network",
    },
    "category": "ftp-default-credentials",
    "description": "The name of the nuclei rule which triggered the finding: ftp-default-credentials",
    "location": "www.example.com",
    "name": "FTP Service with default credentials",
    "severity": "HIGH",
  },
]
`;

exports[`parses secureCodeBox.io result correctly 1`] = `
Array [
  Object {
    "attributes": Object {
      "author": null,
      "extracted_results": null,
      "ip": null,
      "matched": "www.secureCodeBox.io",
      "matcher_name": "github",
      "metadata": null,
      "reference": null,
      "tags": null,
      "timestamp": "2021-12-21T13:48:22.3877066Z",
      "type": "dns",
    },
    "category": "cname-service-detection",
    "description": "The name of the nuclei rule which triggered the finding: cname-service-detection",
    "location": "www.secureCodeBox.io",
    "name": "cname service detection",
    "severity": "INFORMATIONAL",
  },
]
`;

exports[`parses the dns result correctly 1`] = `
Array [
  Object {
    "attributes": Object {
      "author": null,
      "extracted_results": Array [
        "31589 8 1 3490A6806D47F17A34C29E2CE80E8A999FFBE4BE",
        "31589 8 2 CDE0D742D6998AA554A92D890F8184C698CFAC8A26FA59875A990C03E576343C",
        "43547 8 1 B6225AB2CC613E0DCA7962BDC2342EA4F1B56083",
        "43547 8 2 615A64233543F66F44D68933625B17497C89A70E858ED76A2145997EDF96A918",
        "31406 8 1 189968811E6EBA862DD6C209F75623D8D9ED9142",
        "31406 8 2 F78CF3344F72137235098ECBBD08947C2C9001C7F6A085A17F518B5D8F6B916D",
      ],
      "ip": null,
      "matched": "example.com",
      "matcher_name": null,
      "metadata": null,
      "reference": null,
      "tags": null,
      "timestamp": "2021-12-21T14:14:46.5321282Z",
      "type": "dns",
    },
    "category": "dnssec-detection",
    "description": "The name of the nuclei rule which triggered the finding: dnssec-detection",
    "location": "example.com",
    "name": "DNSSEC Detection",
    "severity": "INFORMATIONAL",
  },
]
`;
